backend-rust/src/indexer.rs

```diff
@@ -803,15 +803,28 @@ impl PreparedBlock {
             prepared_block_items
                 .push(PreparedBlockItem::prepare(node_client, data, item_summary, item).await?)
         }
-        let special_items: Vec<_> = data
+        let prepared_account_statements_iter =
+            data.events.iter().zip(data.items.iter()).map(|(item_summary, item)| {
+                PreparedAccountStatement::prepare_from_block_item(
+                    item_summary,
+                    data.block_info.block_height,
+                )
+            });
```

Comment:
I think we will have an issue with this approach of collecting them all in one list like this and then process them.
We need them to be intertwined with the other events when storing to the database. When we `save` these to the database it selects the balance of the account, and therefore if we update the balance from anywhere else when processing the block item, like an in a contract update, we are in trouble.

Reply:
```
        let combined_statements: Vec<_> =
            prepared_account_statements_iter.chain(special_items_iter).collect();
```
I am doing that?